.travis.yml

```diff
@@ -1,35 +1,56 @@
 sudo: required
-dist: xenial
+dist: focal
 language: c
 services:
   - postgresql
   - mysql
+addons:
+  apt:
+    # This has the same effect as running apt-get update, but avoid problems
+    # with signature verification
+    update: true
+  postgresql: '12'
+  mysql: '8'
+  apt:
+    packages:
+      - postgresql-12
+      - mysql-client-8.0
+      - mysql-server-8.0
+env:
+  global:
+    - PGPORT=5433
+    - PGUSER=travis
+    - DBUSER=links
+    - DBNAME=links
 before_install:
   - git diff-index --quiet HEAD --
-  - sudo add-apt-repository -y ppa:avsm/ppa
-  - sudo apt-get -qq update
-  - sudo apt-get install -y ocaml-nox ocaml-native-compilers python3-setuptools python3-pip libev-dev
+  # BEGIN postgresql trust based authentication
+  - sudo sed -i -e '/local.*peer/s/postgres/all/' -e 's/peer\|md5/trust/g' /etc/postgresql/*/main/pg_hba.conf
+  - sudo systemctl restart postgresql
+  # END
+  - sudo apt-get install -y ocaml-nox ocaml-native-compilers opam python3-setuptools python3-pip libev-dev
   - pip3 install -U Sphinx
-  - ./install_local_opam2.sh
-  - ./opam2_local switch create ocaml-base-compiler.4.08.0
-  - eval `./opam2_local config env`
-  - ./opam2_local install conf-libev
-  - ./opam2_local install -y postgresql mysql ounit2
-  - ./tools/install_ocamlformat
+  - opam init --disable-sandboxing -n
+  - opam switch create ocaml-base-compiler.4.08.0
+  - eval `opam config env`
+  - opam install -y conf-libev conf-mysql
+  - opam install -y ocamlformat.0.14.2 ounit2
   - make rule-check
-  - psql -c 'create database links;' -U postgres
-  - mysql -e 'CREATE DATABASE links;'
-  - mysql -e "CREATE USER links IDENTIFIED BY '12345';"
-  - mysql -e 'GRANT ALL ON  links.* TO links;'
+  - psql -c "create database $DBNAME;" -U postgres
+  - psql -c "CREATE ROLE $DBUSER SUPERUSER LOGIN CREATEDB;" -U postgres
+  - mysql -e "CREATE DATABASE $DBNAME;"
+  - mysql -e "CREATE USER $DBUSER IDENTIFIED BY 'links';"
+  - mysql -e "GRANT ALL ON $DBNAME.* TO $DBUSER;"
+  - mysql -e "FLUSH PRIVILEGES;"
 script:
-  - ./opam2_local pin add links . -y
-  - ./opam2_local pin add links-postgresql . -y
-  - ./opam2_local pin add links-sqlite3 . -y
-  - ./opam2_local pin add links-mysql . -y
+  - opam pin add links . -y
+  - opam pin add links-postgresql . -y
+  - opam pin add links-sqlite3 . -y
+  - opam pin add links-mysql . -y
   - make doc
   - make all-ci
   - make tests
-  - ./run-database-tests tests/database -d all
+  - ./run-database-tests tests/database -d mysql
```

Comment:
What is the reason for this change?  If it's just to avoid spending time running the other database tests while getting this PR to work, it should be reverted.

Reply:
good catch! That is supposed to read "all"

Reply:
I changed it to `mysql` for faster turn around when I was debugging the mysql build.